### PR TITLE
1.1-DEV

### PR DESCRIPTION
**Pull request by mssumanth**
Merged 1.0.7 changes to the release candidate of FV1.2 which consists of the following features:
    \* FLOWVISOR-59 : Flowvisor should be able to ask the flow table from the switch
    \* FLOWVISOR-62 : Allow slice perms to be specfied with "rwd" in addition to by number
    \* FLOWVISOR-187: Log level of packets not matching any slice in Flowvisor
    \* FLOWVISOR-15 : LLDP packet made by FlowVisor doesn't have "the end of LLDPDU" nor a TLV header before padding(0xcafebabe)
    \* FLOWVISOR-158: Control planes with multiple flowvisors should not broadcast LLDP packet-ins to all slices

### CODE REVIEW

File: src/org/flowvisor/classifier/FVClassifier.java [hunk 1]
```diff
@@ -1084,9 +1114,84 @@ public void sendFlowStatsResp(FVSlicer fvSlicer, FVStatisticsRequest original) {
...
+		List<Object> params = new ArrayList<Object>();
+		//Serialize the stats to send it via JSON
+		
+		for (int i=0; i<stats.size(); i++){
+			OFFlowStatisticsReply reply = (OFFlowStatisticsReply) stats.get(i);
+			
+			HashMap <String,Object> cache = new HashMap<String,Object>();
+			
+			cache.put(FlowSpace.PRIO, String.valueOf(reply.getPriority()));
+			cache.put(FlowSpace.ACTION, reply.getActions().toString());
```
Comment (alshabib) [inline src/org/flowvisor/classifier/FVClassifier.java]: These should be maps as well. Sending a string is never acceptable.

File: src/org/flowvisor/message/lldp/LLDPTrailer.java [hunk 2]
```diff
@@ -113,32 +188,117 @@ public void appendTo(FVPacketOut po) {
...
+		FVLog.log(LogLevel.DEBUG, null, " OUI: ",packet.get(offset));
+		if(packet.get(offset) != 35) //0x23
+			FVLog.log(LogLevel.ALERT, null, "Wrong OUI1");
+		offset -=1;
+		FVLog.log(LogLevel.DEBUG, null, " OUI: ",packet.get(offset));
+		if(packet.get(offset) != -92) //0xa4 = -92 
+			FVLog.log(LogLevel.ALERT, null, "Wrong OUI2");
+		}
+		catch(IndexOutOfBoundsException ioe){
+			System.out.println("Yikes! The LLDP packet in is not well formed - IndexOutOfBound while getting the trailer ");
```
Comment (alshabib) [inline src/org/flowvisor/message/lldp/LLDPTrailer.java]: This println should go.

**Comment by alshabib** [comment]
Are there tests associated with this code? If yes, where are they?

**Comment by mssumanth** [comment]
I have the tests with changed fv test framework for simpleLLDPPacket
creation with me, will push them
once I incorporate the changes you have mentioned and push them!

**Pull request merged**